src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java

```diff
@@ -2568,13 +2573,13 @@ protected void appendDetail(StringBuffer buffer, String fieldName, Object value)
                 return;
             }
 
-            if (value.getClass() == String.class) {
+            if (value instanceof Number || value.getClass() == Boolean.class) {
 
-                appendValueAsString(buffer, (String)value);
+                buffer.append(value);
                 return;
             }
 
-            buffer.append(value);
+            appendValueAsString(buffer, value.toString());
```

Comment:
Yes. But if `Address` like you suppose here does not *Override* `Object.toString()` (maybe someone just wants Address instance represented by its hash code), then its json style looks like `"address":"Address@1ab2f62"`.